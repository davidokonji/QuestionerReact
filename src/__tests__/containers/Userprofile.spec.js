import React from 'react';
import { shallow } from 'enzyme';
import { UserProfile, mapStateToProps } from '../../containers/userProfile.container';

let wrapper;

describe('User Profile container', () => {
  beforeAll(() => {
    const props = {
      getUser: jest.fn(),
      getComment: jest.fn(),
      getQuestion: jest.fn(),
      getUpcoming: jest.fn()
    };
    wrapper = shallow(<UserProfile {...props} />);
  });

  it('should render component correctlt', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should return map state to props', () => {
    const state = {
      profile: {
        user: {},
        comments: '',
        questionCount: '',
        upcoming: [],
        loading: true
      }
    };
    const action = mapStateToProps(state);

    expect(action).toEqual({
      user: state.profile.user,
      comment: state.profile.comments,
      questionCount: state.profile.questionCount,
      upcoming: state.profile.upcoming,
      loading: state.profile.loading
    });
  });
});
