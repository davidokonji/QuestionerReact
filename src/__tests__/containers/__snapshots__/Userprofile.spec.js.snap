// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Profile container should render component correctlt 1`] = `
<div>
  <withRouter(Connect(NavBar)) />
  <div
    className="container-fluid row mt-5"
  >
    <UserSidebar />
    <UserTabs
      message=""
      status=""
    />
  </div>
  <Footer />
</div>
`;
